File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        boolean[] visited = new boolean[n];
        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(i, n, computers, visited);
                ans++;
            }
        }
        return ans;
    }

    static void dfs(int v, int n, int[][] computers, boolean[] visited) {
        visited[v] = true;
        for (int i = 0; i < n; i++)
            if (computers[v][i] == 1 && !visited[i])
                dfs(i, n, computers, visited);
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

int solution(int n, vector<vector<int>> computers) {
    // computers는 인접행렬인데 이걸 인접리스트처럼 생각해서 (0이아닌경우 j를꺼냄) 타고타고가기
    int answer = 0;
    bool visited[201]={};
    for(int i=0; i<n; i++){
        if(visited[i]) continue;
        answer++;
        queue<int> q;
        q.push(i);
        visited[i]=true;
        while(!q.empty()){
            int c=q.front();
            q.pop();
            for(int j=0; j<n; j++){
                if(computers[c][j]==1 && !visited[j]) { // i->j
                    visited[j]=true;
                    q.push(j);
                }
            }
        }
    }
    return answer;
}